osmomath/exp2.go

```diff
@@ -0,0 +1,94 @@
+package osmomath
+
+import "fmt"
+
+var (
+	// Truncated at precision end.
+	// See scripts/approximations/main.py exponent_approximation_choice function for details.
+	numeratorCoefficients13Param = []BigDec{
+		MustNewDecFromStr("1.000000000000000000000044212244679434"),
+		MustNewDecFromStr("0.352032455817400196452603772766844426"),
+		MustNewDecFromStr("0.056507868883666405413116800969512484"),
+		MustNewDecFromStr("0.005343900728213034434757419480319916"),
+		MustNewDecFromStr("0.000317708814342353603087543715930732"),
+		MustNewDecFromStr("0.000011429747507407623028722262874632"),
+		MustNewDecFromStr("0.000000198381965651614980168744540366"),
+	}
+
+	// Rounded up at precision end.
+	// See scripts/approximations/main.py exponent_approximation_choice function for details.
+	denominatorCoefficients13Param = []BigDec{
+		OneDec(),
+		MustNewDecFromStr("0.341114724742545112949699755780593311").Neg(),
+		MustNewDecFromStr("0.052724071627342653404436933178482287"),
+		MustNewDecFromStr("0.004760950735524957576233524801866342").Neg(),
+		MustNewDecFromStr("0.000267168475410566529819971616894193"),
+		MustNewDecFromStr("0.000008923715368802211181557353097439").Neg(),
+		MustNewDecFromStr("0.000000140277233177373698516010555916"),
+	}
+
+	// maxSupportedExponent = 2^10. The value is chosen by benchmarking
+	// when the underlying internal functions overflow.
+	// If needed in the future, Exp2 can be reimplemented to allow for greater exponents.
+	maxSupportedExponent = MustNewDecFromStr("2").PowerInteger(9)
+)
+
+// Exp2 takes 2 to the power of a given non-negative decimal exponent
+// and returns the result.
+// The computation is performed by using th following property:
+// 2^decimal_exp = 2^{integer_exp + fractional_exp} = 2^integer_exp * 2^fractional_exp
+// The max supported exponent is defined by the global maxSupportedExponent.
+// If a greater exponent is given, the function panics.
+// Panics if the exponent is negative.
+func Exp2(exponent BigDec) BigDec {
+	if exponent.Abs().GT(maxSupportedExponent) {
+		panic(fmt.Sprintf("integer exponent %s is too large, max (%s)", exponent, maxSupportedExponent))
+	}
+	if exponent.IsNegative() {
+		panic(fmt.Sprintf("negative exponent %s is not supported", exponent))
+	}
+
+	integerExponentDec := exponent.TruncateDec()
+	integerExponent := integerExponentDec.TruncateInt()
+	integerResult := twoBigDec.PowerInteger(integerExponent.Uint64())
```

Comment:
```suggestion
	// TODO: Switch the application of the integer result to just being a bit shift
	integerResult := twoBigDec.PowerInteger(integerExponent.Uint64())
```

Reply:
This would mean we also have to manually do bounds checks, right?

Reply:
Shouldn't _have_ to, it should be implied by the max exponent check further up

Reply:
Great idea. Implemented it right away. Yes, this should be safe because of the check above